src/python/pants/backend/python/macros/poetry_requirements_test.py

```diff
@@ -81,7 +100,49 @@ def test_max_tilde(test, exp) -> None:
     ],
 )
 def test_handle_str(test, exp) -> None:
-    assert parse_str_version("foo", test, "") == f"foo {exp}"
+    assert parse_str_version(test, proj_name="foo", file_path="", extras_str="") == f"foo {exp}"
+
+
+def test_add_markers() -> None:
+
+    attr_mark = PyprojectAttr({"markers": "platform_python_implementation == 'CPython'"})
```

Comment:
I think this should work:

```suggestion
    attr_mark = {"markers": "platform_python_implementation == 'CPython'"}
```

If it doesn't, that's fine to keep as before. But generally, the idea of TypedDict is that the actual concrete type is still a `dict`, only you are saying what keys are expected.

--

If this does work, it'd be great to simplify the rest of the callsites in this file.

Reply:
This does not work unfortunately due to the heterogeneity of the dictionary: add_markers has to be passed the new type so `extras` can be of type `list` which isn't permitted with a regular `dict` type.